crates/bevy_render/src/render_resource/bind_group_layout_entries.rs

```diff
@@ -0,0 +1,582 @@
+        &self.entries
+    }
+}
+
+pub mod binding_types {
+    use crate::render_resource::{
+        BufferBindingType, SamplerBindingType, TextureSampleType, TextureViewDimension,
+    };
+    use std::num::NonZeroU64;
+    use wgpu::{BindingType, StorageTextureAccess, TextureFormat};
+
+    use super::*;
+
+    #[allow(unused)]
+    pub fn storage_buffer(
+        has_dynamic_offset: bool,
+        min_binding_size: Option<NonZeroU64>,
+    ) -> BindGroupLayoutEntryBuilder {
+        BindingType::Buffer {
+            ty: BufferBindingType::Storage { read_only: false },
+            has_dynamic_offset,
+            min_binding_size,
+        }
+        .into_bind_group_layout_entry_builder()
+    }
+
+    #[allow(unused)]
+    pub fn storage_buffer_read_only(
+        has_dynamic_offset: bool,
+        min_binding_size: Option<NonZeroU64>,
+    ) -> BindGroupLayoutEntryBuilder {
+        BindingType::Buffer {
+            ty: BufferBindingType::Storage { read_only: true },
+            has_dynamic_offset,
+            min_binding_size,
+        }
+        .into_bind_group_layout_entry_builder()
+    }
+
+    #[allow(unused)]
+    pub fn uniform_buffer(
+        has_dynamic_offset: bool,
+        min_binding_size: Option<NonZeroU64>,
+    ) -> BindGroupLayoutEntryBuilder {
+        BindingType::Buffer {
+            ty: BufferBindingType::Uniform,
+            has_dynamic_offset,
+            min_binding_size,
+        }
+        .into_bind_group_layout_entry_builder()
+    }
+
+    #[allow(unused)]
+    pub fn texture_2d(sample_type: TextureSampleType) -> BindGroupLayoutEntryBuilder {
+        BindingType::Texture {
+            sample_type,
+            view_dimension: TextureViewDimension::D2,
+            multisampled: false,
+        }
+        .into_bind_group_layout_entry_builder()
+    }
+
+    #[allow(unused)]
+    pub fn texture_2d_multisampled(sample_type: TextureSampleType) -> BindGroupLayoutEntryBuilder {
+        BindingType::Texture {
+            sample_type,
+            view_dimension: TextureViewDimension::D2,
+            multisampled: true,
+        }
+        .into_bind_group_layout_entry_builder()
+    }
+
+    #[allow(unused)]
+    pub fn texture_2d_array(sample_type: TextureSampleType) -> BindGroupLayoutEntryBuilder {
+        BindingType::Texture {
+            sample_type,
+            view_dimension: TextureViewDimension::D2Array,
+            multisampled: false,
+        }
+        .into_bind_group_layout_entry_builder()
+    }
+
+    #[allow(unused)]
+    pub fn texture_2d_array_multisampled(
+        sample_type: TextureSampleType,
+    ) -> BindGroupLayoutEntryBuilder {
+        BindingType::Texture {
+            sample_type,
+            view_dimension: TextureViewDimension::D2Array,
+            multisampled: true,
+        }
+        .into_bind_group_layout_entry_builder()
+    }
+
+    #[allow(unused)]
+    pub fn texture_2d_f32(filterable: bool) -> BindGroupLayoutEntryBuilder {
```

Comment:
i think this is only f32 if the bound texture format is f32, it could also be f16 (although currently unsupported in wgpu)? not 100% sure about that though.

same for uint/u32 and int/i32 as well.

Reply:
Right now wgpu only supports f32/u32/i32 which is why I made it like that. If they ever had more size we'll just specialilze this one to actually be f32 and introduce a new one for other sizes.

Reply:
Although, that's part of why using just `texture_2d()` and let people specify the type is a decent solution too.

Reply:
what is the right binding type for `Rgba16Float`? surely still Float?

Reply:
Yes, but as far as I know you can't have an `f16` in wgsl (that may have changed, it's been a while since I checked), but I think I'll just rename it to float/uint/sint to keep the wgpu naming and avoid sizing issues since size is not part of this api anyway.

Reply:
I remembered some sizes not being supported, but apparently f16 was a thing for a while. Not sure what I was thinking about honestly 😅